snuba/writer.py

```diff
@@ -58,8 +59,16 @@ def __encode(self, row):
 
     def write(self, rows):
         parameters = self.__options.copy()
-        parameters['query'] = "INSERT INTO {table} FORMAT JSONEachRow".format(table=self.__schema.get_table_name())
-        requests.post(
-            urljoin(self.__base_url, '?' + urlencode(parameters)),
-            data=map(self.__encode, rows),
-        ).raise_for_status()
+        parameters['query'] = f"INSERT INTO {self.__schema.get_table_name()} FORMAT JSONEachRow"
+        resp = self.__pool.urlopen(
+            'POST',
+            '/?' + urlencode(parameters),
+            headers={
+                'Connection': 'keep-alive',
+                'Accept-Encoding': 'gzip,deflate',
+            },
+            body=map(self.__encode, rows),
+            chunked=True,
+        )
+        if resp.status//100 != 2:
```

Comment:
If you know what the actual status code is, I'd prefer we check against that. `raise_for_status` was significantly more lenient than this, so happy to do whatever makes sense for you.

Reply:
It does not matter much right now, we need to revisit this code anyway to surface the actual error message no matter what the error code is so "raise_for_status" has to be removed anyway.